### PR TITLE
More SAI

### PR DESCRIPTION
**Pull request by Gultask**
Don't merge yet, need to test

### CODE REVIEW

File: src/app/shared/modules/sai-editor/sai-comment-generator.service.ts [hunk 1]
```diff
@@ -447,6 +453,14 @@ export class SaiCommentGeneratorService {
         actionLine = actionLine.replace('_enableDisableActionParamOne_', 'Enable');
       }
     }
+	
+	if (actionLine.indexOf('_enableDisableInvertActionParamOne_') > -1) {
+      if (`${smartScript.action_param1}` === '0') {
+        actionLine = actionLine.replace('_enableDisableInvertActionParamOne_', 'Enable');
+      } else {
+        actionLine = actionLine.replace('_enableDisableInvertActionParamOne_', 'Disable');
+      }
```
Comment (Helias) [inline src/app/shared/modules/sai-editor/sai-comment-generator.service.ts:462]:
```suggestion
      const enableOrDisable = `${smartScript.action_param1}` === '0' ? 'Enable' : 'Disable';
      actionLine = actionLine.replace('_enableDisableInvertActionParamOne_', enableOrDisable);
```

File: src/app/shared/modules/sai-editor/sai-comment-generator.service.ts [hunk 2]
```diff
@@ -475,6 +489,55 @@ export class SaiCommentGeneratorService {
...
+        case 0:
+          actionLine = actionLine.replace('_waypointStartActionParamThree_', 'Waypoint ');
+          break;
+        case 1:
+          actionLine = actionLine.replace('_waypointStartActionParamThree_', 'Patrol ');
+          break;
+        default:
+          actionLine = actionLine.replace('_waypointStartActionParamThree_', '[Unknown Value]');
+          break;
+      }
```
Comment (Helias) [inline src/app/shared/modules/sai-editor/sai-comment-generator.service.ts:504]: The only part that changes can be isolated to reduce the redundancy.
```suggestion
      let waypointReplace: string;
      switch (Number(smartScript.action_param1)) {
        case 0:
          waypointReplace = 'Waypoint ';
          break;
        case 1:
          waypointReplace = 'Patrol ';
          break;
        default:
          waypointReplace = '[Unknown Value]';
          break;
      }

      actionLine = actionLine.replace('_waypointStartActionParamThree_', waypointReplace);
```

**Comment by Helias** [comment]
ignore the macos CI, it's a bit bugged

**Comment by Gultask** [comment]
I don't really know how to compile it yet, sorry :(
I have an idea to use the waypoint name instead of just the numbers, so I'll see what I can do with this

**Comment by Gultask** [comment]
Should be done now

**Comment by Helias** [comment]
I will check why the pipeline is failing and fix the unit-tests

**Pull request merged**